projects/js-packages/publicize-components/src/components/social-post-modal/modal.tsx

```diff
@@ -15,7 +16,19 @@ import styles from './styles.module.scss';
  * @return {import('react').ReactNode} - Social Post Modal component.
  */
 export function SocialPostModal() {
-	const [ isModalOpen, toggleModal ] = useReducer( state => ! state, false );
+	const [ isModalOpen, toggleModal ] = useReducer(
+		state => {
+			const url = new URL( window.location.href );
+			// If the modal is opened with the share post query arg, remove it from the URL.
+			if ( state && url.searchParams.has( 'jetpackSidebar' ) ) {
+				url.searchParams.delete( 'jetpackSidebar' );
+				window.history.replaceState( null, '', url.toString() );
+			}
+			return ! state;
```

Comment:
Do we have to remove it? It's probably cleaner, but it would be better if this didn't live here. I also think it's fine to leave it in the URL so if the page is reloaded, it remains open

Reply:
If we don't remove it, switching to some other settings sidebar and switching back to Jetpack will open the modal again, which can be annoying.

Reply:
That points further to the state being in the wrong place. I was thinking that the initial state of the modal should be passed in and that we shouldn't be coupling the modal to this queryArg logic.